app.js

```diff
@@ -38,6 +80,74 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+app.get('/fruits', (req, res, next) => {
+  const { name } = req.query
+  const fruit = fruits.filter(f => !name || f.name.includes(name))
+  res.json(fruit)
+  next( {status: 200} )
+})
+
+app.get('/fruits/:id', (req, res, next) => {
+  const { fruits } = data
+  const { id } = req.params
+  const fruit = fruits.find(f => f.id === id)
+
+  if (!fruit) {
+    const message = `Could not find fruit with ID of ${id}`
+    next({ status: 404, message })
+  }
+  res.json(fruit)
+  next( {status: 200})
```

Comment:
I think you're misunderstanding how `next()` works. If you put something in the `next()` function, it will move along to the error handler. If you want to respond with a particular status code, you would do so like the following:
```js
res.status(200).json(fruit)
```

Reply:
Additionally, if two or more responses are sent, you'll get the following error:
```
Cannot set headers after they are sent to the client
```